src/Export/SalesChannelService.php

```diff
@@ -32,26 +33,18 @@ public function __construct(
      * assigned to any sales channel.
      */
     public function getSalesChannelContext(
-        ?SalesChannelContext $currentContext,
+        SalesChannelContext $currentContext,
         string $shopkey
     ): ?SalesChannelContext {
         $systemConfigEntities = $this->systemConfigRepository->search(
             (new Criteria())->addFilter(new EqualsFilter('configurationKey', 'FinSearch.config.shopkey')),
             $currentContext->getContext()
         );
 
-        /** @var SystemConfigEntity $systemConfigEntity */
+        /** @var FinSearchConfigEntity $systemConfigEntity */
         foreach ($systemConfigEntities as $systemConfigEntity) {
             if ($systemConfigEntity->getConfigurationValue() === $shopkey) {
-                // When the configured sales channel id is null, the shopkey is configured for all sales channels.
-                if ($systemConfigEntity->getSalesChannelId() === null) {
-                    return $currentContext;
-                }
-
-                return $this->salesChannelContextFactory->create(
-                    $currentContext->getToken(),
-                    $systemConfigEntity->getSalesChannelId()
-                );
+                return $currentContext;
```

Comment:
Why was this changed? The exported language should be based on the given shopkey. Please elaborate.

Reply:
The condition before the return checks for the shopkey given if the sales channel has a same shopkey or not. SInce sales channel is required for shopkey in this implementation.

Reply:
The used sales channel should not matter. We need to get the sales channel, based on the shopkey. The only exception before this was "All Sales Channel", because if it has been configured for all Sales Channels, it did not matter. Since you now no longer can configure a shopkey for all sales channels, this condition can be removed entirely.

https://german-shop.de/findologic=shopkey=german-shopkey => Must return products in German translation.
https://german-shop.de/findologic=shopkey=english-shopkey => Must return products in English translation.
etc.